src/ccs/cccs.rs

```diff
@@ -95,37 +112,44 @@ impl<G: Group> CCCSShape<G> {
   /// polynomial over x
   pub fn compute_Q(
     &self,
-    z: &Vec<G::Scalar>,
+    ccs: &CCS<G>,
+    ccs_mles: &[MultilinearPolynomial<G::Scalar>],
+    ck: &CommitmentKey<G>,
     beta: &[G::Scalar],
   ) -> Result<VirtualPolynomial<G::Scalar>, NovaError> {
-    let q = self.compute_q(z)?;
+    let q = self.compute_q(ccs, ccs_mles)?;
     q.build_f_hat(beta)
   }
 
   /// Perform the check of the CCCS instance described at section 4.1
   pub fn is_sat(
     &self,
+    ccs: &CCS<G>,
+    ccs_mles: &[MultilinearPolynomial<G::Scalar>],
     ck: &CommitmentKey<G>,
-    w: &CCSWitness<G>,
-    x: &CCCSInstance<G>,
   ) -> Result<(), NovaError> {
     // check that C is the commitment of w. Notice that this is not verifying a Pedersen
     // opening, but checking that the Commmitment comes from committing to the witness.
-    assert_eq!(x.C, CE::<G>::commit(ck, &w.w));
-
-    // check CCCS relation
-    let z: Vec<G::Scalar> = [vec![G::Scalar::ONE], x.x.clone(), w.w.to_vec()].concat();
+    assert_eq!(self.w_comm, CE::<G>::commit(ck, &self.z[(1 + ccs.l)..]));
 
     // A CCCS relation is satisfied if the q(x) multivariate polynomial evaluates to zero in the hypercube
-    let q_x = self.compute_q(&z).unwrap();
-    for x in BooleanHypercube::new(self.ccs.s) {
+    let q_x = self.compute_q(ccs, ccs_mles).unwrap();
+    for x in BooleanHypercube::new(ccs.s) {
       if !q_x.evaluate(&x).unwrap().is_zero().unwrap_u8() == 0 {
         return Err(NovaError::UnSat);
       }
     }
 
     Ok(())
   }
+
+  // XXX: Pending to add a commit fn:
```

Comment:
So is this a TODO?

Reply:
No, as the commitment is computed on the CCCS generation.

Reply:
Fixed in 52f6a62